### ISSUE TITLE
Change is_training of tf.contrib.layers.batch_norm to conditional function

### ISSUE THREAD

**Issue by carpedm20** [tensorflow/tensorflow#3265]
`tf.contrib.layers.batch_norm` is using `is_training` as a python boolean variable so I may have to define two different ops which share its variables by using `reuse=True` to the second op.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers.py#L209

However, I think it is better to use `tf.placeholder` or other tensor variables as `is_training` and use `tf.cond` to dynamically change training and testing phase without defining two different ops. Or is there any better usage of using `tf.contrib.layers.batch_norm` that I couldn't think of? Also, is `tf.contrib.layers.fully_connected` ready to use `batch_norm` as a `normalizer_fn` (because I guess this is still in contrib for several months)?

**Comment by pawni** [comment]
I am currently using it like that with my own wrapper around it and think that it is supposed to work like that as `tf.contrib.layers.dropout` does: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers.py#L433-L435

**Comment by jmchen-g** [comment]
@ilblackdragon Could you take a look at this please?

**Comment by sguada** [comment]
I already have done it internally, it will push soon.

**Comment by sguada** [comment]
Fixed in https://github.com/tensorflow/tensorflow/commit/9da5fc8e6425cabd61fc36f0dcc1823a093d5c1d#diff-94bbcef0ec8a5cdef55f705e99c2b2ed

**Comment by brando90** [comment]
do we need to update tensorflow? I tried but it seems it made no difference.

**Comment by sguada** [comment]
Yeah you need 0.10 https://github.com/tensorflow/tensorflow/tree/r0.10
